src/plugin/ipc/socket/socketconnection.cpp

```diff
@@ -418,6 +418,57 @@ TcpConnection::TcpConnection(const TcpConnection &parent,
 }
 
 void
+TcpConnection::sendPeerInformation()
+{
+  struct sockaddr_storage local, remote;
+  socklen_t localsz, remotesz;
+  switch (_type) {
+  case TCP_CONNECT:
+  {
+    localsz = sizeof(local);
+    JASSERT(getsockname(_fds[0],
+                        (struct sockaddr *)&local,
+                        &localsz) == 0);
+    remotesz = _connectAddrlen;
+    memcpy(&remote, &_connectAddr, sizeof(remotesz));
+    break;
+  }
+  case TCP_ACCEPT:
+  {
+    localsz = _bindAddrlen;
+    memcpy(&local, &_bindAddr, sizeof(localsz));
+    remotesz = sizeof(remote);
+    JASSERT(getpeername(_fds[0],
+                        (struct sockaddr *)&remote,
+                        &remotesz) == 0);
+    break;
+  }
+  default:
+    break;
+  }
+  dmtcp_send_key_val_pair_to_coordinator("RemoteConnections",
+                                         &local, localsz,
+                                         &remote, remotesz);
+}
+
+void
+TcpConnection::recvPeerInformation()
+{
+  struct sockaddr local, remote;
+  socklen_t localsz = sizeof(local), remotesz;
+  if (_type == TCP_CONNECT) {
+    int ret = dmtcp_send_query_to_coordinator("RemoteConnections",
+                                              &local, localsz,
+                                              &remote, &remotesz);
+    if (ret != 0) {
+      JASSERT(remotesz == sizeof(remote));
+    } else {
+      markExternalConnect();
+    }
+  }
+}
```

Comment:
What's the key used here? it seems `local` is not initialized. Also, should we do the same for accepted sockets?